lib/hound/session_server.ex

```diff
@@ -22,6 +22,17 @@ defmodule Hound.SessionServer do
     end
   end
 
+  def current_session_name(pid) do
+    case :ets.lookup(@name, pid) do
+      [{^pid, _ref, session_id, all_sessions}] ->
+        Enum.find_value all_sessions, fn
+          {name, ^session_id} -> name
```

Comment:
Sorry for the false alert, I just checked and it seems that this is only allowed from 1.2: https://github.com/elixir-lang/elixir/blob/v1.2/CHANGELOG.md#v120-2016-01-01

> [Kernel] Allow the pin operator ^ in fn clauses and on the left side of <- in for comprehensions

Sorry for the extra work, but could you revert to the guard please?